Fix golint errors in tchannel package

```
> golint . | grep -v _string.go
frame.go:78:1: exported method FrameHeader.MarshalJSON should have comment or be unexported
logger.go:76:1: exported function ErrField should have comment or be unexported
peer_heap.go:85:31: exported method PopPeer returns unexported type *tchannel.peerScore, which can be annoying to use
```

We may want to add a list of packages we check (or check all of them, and have ignore rules e.g. for _string.go files)

--- a/.gitignore
+++ b/.gitignore
@@ -2,6 +2,9 @@ build
 Godeps/_workspace
 thrift-gen-release/
 
+# Lint output
+lint.log
+
 # Cover profiles
 *.out
 

--- a/Makefile
+++ b/Makefile
@@ -90,10 +90,16 @@ cover: cover_profile
 cover_ci: cover_profile
 	goveralls -coverprofile=$(BUILD)/coverage.out -service=travis-ci
 
-vet:
-	echo Vetting packages for potential issues...
-	go tool vet $(PKGS)
-	echo
+
+FILTER := grep -v -e '_string.go' -e '/gen-go/' -e '/mocks/'
+lint:
+	@echo "Running golint"
+	-golint ./... | $(FILTER) | tee lint.log
+	@echo "Running go vet"
+	-go tool vet $(PKGS) | tee -a lint.log
+	@echo "Checking gofmt"
+	-gofmt -d . | tee -a lint.log
+	@[ ! -s lint.log ]
 
 thrift_example: thrift_gen
 	go build -o $(BUILD)/examples/thrift       ./examples/thrift/main.go
@@ -125,5 +131,5 @@ release_thrift_gen: clean setup
 	tar -czf thrift-gen-release.tar.gz $(THRIFT_GEN_RELEASE)
 	mv thrift-gen-release.tar.gz $(THRIFT_GEN_RELEASE)/
 
-.PHONY: all help clean fmt format get_thrift install install_ci release_thrift_gen packages_test test test_ci vet
-.SILENT: all help clean fmt format test vet
+.PHONY: all help clean fmt format get_thrift install install_ci release_thrift_gen packages_test test test_ci lint
+.SILENT: all help clean fmt format test lint

--- a/examples/ping/main.go
+++ b/examples/ping/main.go
@@ -62,7 +62,7 @@ func listenAndHandle(s *tchannel.Channel, hostPort string) {
 	// If no error is returned, the listen was successful. Serving happens in the background.
 	if err := s.ListenAndServe(hostPort); err != nil {
 		log.WithFields(
-			tchannel.LogField{"hostPort", hostPort},
+			tchannel.LogField{Key: "hostPort", Value: hostPort},
 			tchannel.ErrField(err),
 		).Fatal("Couldn't listen.")
 	}

--- a/hyperbahn/client.go
+++ b/hyperbahn/client.go
@@ -59,6 +59,8 @@ const (
 
 const hyperbahnServiceName = "hyperbahn"
 
+// UnmarshalYAML implements the yaml.UnmarshalYAML interface. This allows FailStrategy
+// to be specified as a string (e.g. "fatal") in configuration files.
 func (f *FailStrategy) UnmarshalYAML(unmarshal func(v interface{}) error) error {
 	var strategy string
 	if err := unmarshal(&strategy); err != nil {

--- a/pprof/pprof.go
+++ b/pprof/pprof.go
@@ -43,7 +43,7 @@ func Register(registrar tchannel.Registrar) {
 		req, err := thttp.ReadRequest(call)
 		if err != nil {
 			registrar.Logger().WithFields(
-				tchannel.LogField{"err", err.Error()},
+				tchannel.LogField{Key: "err", Value: err.Error()},
 			).Warn("Failed to read HTTP request.")
 			return
 		}

--- a/raw/handler.go
+++ b/raw/handler.go
@@ -35,11 +35,11 @@ type Handler interface {
 
 // Args parses the arguments from an incoming call req.
 type Args struct {
-	Caller    string
-	Format    tchannel.Format
+	Caller string
+	Format tchannel.Format
 	Method string
-	Arg2      []byte
-	Arg3      []byte
+	Arg2   []byte
+	Arg3   []byte
 }
 
 // Res represents the response to an incoming call req.

--- a/retry_request_test.go
+++ b/retry_request_test.go
@@ -57,7 +57,7 @@ func TestRequestStateRetry(t *testing.T) {
 		}
 
 		_, err := raw.CallV2(ctx, sc, raw.CArgs{
-			Method:   "echo",
+			Method:      "echo",
 			CallOptions: &CallOptions{RequestState: rs},
 		})
 		return err

--- a/stats_test.go
+++ b/stats_test.go
@@ -190,7 +190,7 @@ func TestStatsWithRetries(t *testing.T) {
 
 				sc := ch.GetSubChannel(serverCh.ServiceName())
 				_, err := raw.CallV2(ctx, sc, raw.CArgs{
-					Method:   "req",
+					Method:      "req",
 					CallOptions: &CallOptions{RequestState: rs},
 				})
 				return err

--- a/testutils/channel.go
+++ b/testutils/channel.go
@@ -80,7 +80,7 @@ type rawFuncHandler struct {
 
 func (h rawFuncHandler) OnError(ctx context.Context, err error) {
 	h.ch.Logger().WithFields(
-		tchannel.LogField{"context", ctx},
+		tchannel.LogField{Key: "context", Value: ctx},
 		tchannel.ErrField(err),
 	).Error("simpleHandler OnError.")
 }

